spec/models/validations/shared_validations_spec.rb

```diff
@@ -215,4 +215,30 @@
       end
     end
   end
+
+  describe "validate numeric question input" do
+    it "does not allow letters" do
+      sales_log.income1 = "abc"
+      shared_validator.validate_numeric_input(sales_log)
+      expect(sales_log.errors[:income1]).to include I18n.t("validations.numeric.format", field: "Buyer 1’s gross annual income")
+    end
+
+    it "does not allow special characters" do
+      sales_log.income1 = "3%5"
+      shared_validator.validate_numeric_input(sales_log)
+      expect(sales_log.errors[:income1]).to include I18n.t("validations.numeric.format", field: "Buyer 1’s gross annual income")
+    end
+
+    it "allows a digit" do
+      sales_log.income1 = "300"
+      shared_validator.validate_numeric_input(sales_log)
+      expect(sales_log.errors[:income1]).to be_empty
+    end
+
+    it "allows a decimal point" do
+      sales_log.income1 = "300.78"
+      shared_validator.validate_numeric_input(sales_log)
+      expect(sales_log.errors[:income1]).to be_empty
+    end
```

Comment:
Does this want a test around having the right characters but not in the right format (so e.g. not "300." or "3.0.0")?

(Might be getting a bit fussy to have too many cases though)

Reply:
Yup, added now!